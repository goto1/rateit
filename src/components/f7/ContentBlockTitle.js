import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`margin-top: 25px !important;`;

const ContentBlockTitle = ({ children }) =>
  <Wrapper className="content-block-title">
    {children}
  </Wrapper>;

ContentBlockTitle.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
};

export default ContentBlockTitle;
