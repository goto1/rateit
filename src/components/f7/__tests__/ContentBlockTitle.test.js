import React from "react";
import ContentBlockTitle from "../ContentBlockTitle";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";
import "jest-styled-components";

describe("ContentBlockTitle component", () => {
  const tree = renderer
    .create(<ContentBlockTitle>HelloWorld!</ContentBlockTitle>)
    .toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it("renders with a style rule `margin-top` equal to 25px", () => {
    expect(tree).toHaveStyleRule("margin-top", "25px !important");
  });

  it("contains `HelloWorld` component", () => {
    const HelloWorld = <div>HelloWorld!</div>;
    const wrapper = shallow(
      <ContentBlockTitle>
        {HelloWorld}
      </ContentBlockTitle>
    );
    expect(wrapper).toContainReact(HelloWorld);
  });
});
