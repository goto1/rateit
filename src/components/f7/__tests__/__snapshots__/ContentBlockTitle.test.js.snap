// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlockTitle component renders correctly 1`] = `
.ewlKZX {
  margin-top: 25px !important;
}

<div
  className="content-block-title sc-bdVaJa ewlKZX"
>
  HelloWorld!
</div>
`;
